qemu/tests/virtio_fs_share_data.py

```diff
@@ -0,0 +1,96 @@
+import logging
+import os
+
+from avocado.utils import process
+
+from virttest import data_dir
+from virttest import error_context
+from virttest import utils_disk
+from virttest import utils_misc
+from virttest.remote import scp_to_remote
+
+from provider.storage_benchmark import generate_instance
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test virtio-fs by sharing the data between host and guest.
+    Steps:
+        1. Create shared directories on the host.
+        2. Run virtiofsd daemons on the host.
+        3. Boot a guest on the host with virtiofs options.
+        4. Log into guest then mount the virtiofs targets.
+        5. Generate files or run stress on the mount points inside guest.
+
+    :param test: QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param env: Dictionary with test environment.
+    """
+    cmd_dd = params.get('cmd_dd')
+    cmd_md5 = params.get('cmd_md5')
+
+    cmd_pjdfstest = params.get('cmd_pjdfstest')
+    pjdfstest_pkg = params.get('pjdfstest_pkg')
+
+    fio_options = params.get('fio_options')
+    io_timeout = params.get_numeric('io_timeout')
+
+    username = params.get('username')
+    password = params.get('password')
+    port = params.get('file_transfer_port')
+
+    vm = env.get_vm(params.get("main_vm"))
+    vm.verify_alive()
+    session = vm.wait_for_login()
+    host_addr = vm.get_address()
+
+    for fs in params.objects("filesystems"):
+        fs_params = params.object_params(fs)
+        fs_target = fs_params.get("fs_target")
+        fs_dest = fs_params.get("fs_dest")
+
+        fs_source = fs_params.get("fs_source_dir")
+        base_dir = fs_params.get("fs_source_base_dir", data_dir.get_data_dir())
+        if not os.path.isabs(fs_source):
+            fs_source = os.path.join(base_dir, fs_source)
+        guest_data = os.path.join(fs_dest, 'fs_test')
+        host_data = os.path.join(fs_source, 'fs_test')
+
+        error_context.context("Create a destination directory %s "
+                              "inside guest." % fs_dest, logging.info)
+        utils_misc.make_dirs(fs_dest, session)
+
+        error_context.context("Mount virtiofs target %s to %s inside guest."
+                              % (fs_target, fs_dest), logging.info)
+        utils_disk.mount(fs_target, fs_dest, 'virtiofs', session=session)
+
```

Comment:
it better to clean the host/guest directories created, that helps to start each test with fresh env.

Reply:
The guest directories will be clean automatically after the guest shutdown since the directories are created in the `/mnt` where mounted the virtio-fs target to, the guest will umount it during shutting down. Although the data still in the shared directories on the host, the `env_process` will remove them.